azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -0,0 +1,360 @@
+        rg: Optional[str] = None,
+    ):
+        super(MessageEndpoint, self).__init__(cmd, hub_name, rg)
+
+    def create(
+        self,
+        endpoint_name: str,
+        endpoint_type: str,
+        endpoint_account_name: Optional[str] = None,
+        endpoint_resource_group: Optional[str] = None,
+        endpoint_subscription_id: Optional[str] = None,
+        endpoint_policy_name: Optional[str] = None,
+        connection_string: Optional[str] = None,
+        container_name: Optional[str] = None,
+        encoding: Optional[str] = None,
+        batch_frequency: int = 300,
+        chunk_size_window: int = 300,
+        file_name_format: str = '{iothub}/{partition}/{YYYY}/{MM}/{DD}/{HH}/{mm}',
+        authentication_type: Optional[str] = None,
+        endpoint_uri: Optional[str] = None,
+        entity_path: Optional[str] = None,
+        collection_name: Optional[str] = None,
+        database_name: Optional[str] = None,
+        primary_key: Optional[str] = None,
+        secondary_key: Optional[str] = None,
+        partition_key_name: Optional[str] = None,
+        partition_key_template: Optional[str] = None,
+        identity: Optional[str] = None
+    ):
+        resource_group_name = self.hub_resource.additional_properties['resourcegroup']
+        if not endpoint_resource_group:
+            endpoint_resource_group = resource_group_name
+        if not endpoint_subscription_id:
+            endpoint_subscription_id = self.hub_resource.additional_properties['subscriptionid']
+
+        endpoint_identity = None
+        if identity and authentication_type != AuthenticationType.IdentityBased.value:
+            raise ArgumentUsageError(
+                "In order to use an identity for authentication, you must select --auth-type as 'identityBased'."
+            )
+        elif identity and identity not in [IdentityType.none.value, SYSTEM_ASSIGNED_IDENTITY]:
+            endpoint_identity = ManagedIdentity(
+                user_assigned_identity=identity
+            )
+        elif authentication_type != AuthenticationType.IdentityBased.value and not connection_string:
+            # check for connection string args
+            error_msg = "Please provide a connection string '--connection-string/-c'"
+            if endpoint_type.lower() in [
+                EndpointType.EventHub.value, EndpointType.ServiceBusQueue.value, EndpointType.ServiceBusTopic.value
+            ]:
+                raise ArgumentUsageError(
+                    error_msg + " or endpoint namespace '--endpoint-namespace', endpoint "
+                    "entity path '--entity-path', and policy name '--policy-name'."
+                )
+            else:
+                raise ArgumentUsageError(
+                    error_msg + " or endpoint account '--endpoint-account'."
+                )
+
+        endpoints = self.hub_resource.properties.routing.endpoints
+        if EndpointType.EventHub.value == endpoint_type.lower():
```

Comment:
I hate the length of this command, is there a better way to refactor it or use helper functions

Reply:
Just off the top of my head, a few things to simplify:
- repetitive checks of `if authentication_type != AuthenticationType.IdentityBased.value and not connection_string:` could be resolved to a single variable like `should_create_cstring` (that name is probably bad but anyway)
- repetitive `append` with custom objects for a few specific endpoint types (all using the same attribute), could possibly create the endpoint object once and use it in a few places:
  ````
  new_endpoint = {
        "connectionString": connection_string,
        "name": endpoint_name,
        "subscriptionId": endpoint_subscription_id,
        "resourceGroup": endpoint_resource_group,
        "authenticationType": authentication_type,
        "endpointUri": endpoint_uri,
        "entityPath": entity_path,
        "identity": endpoint_identity
   }
  ...
  endpoints.event_hubs.append(new_endpoint)
  ...
  endpoints.service_bus_queues.append(new_endpoint)
  ...etc

Reply:
did some refactoring

Reply:
Seems a bit cleaner (duplicate logic removed) - unsure why some of these branches of the logic create a `connection_string` value thought that's not used... are these supposed to be part of the `new_endpoint` object like `new_endpoint["connectionString"]`?

Reply:
fixed